.github/workflows/ci.yml

```diff
@@ -49,13 +88,73 @@ jobs:
           key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
 
       - name: Check that workflows are up to date
-        run: sbt ++${{ matrix.scala }} githubWorkflowCheck
+        run: sbt githubWorkflowCheck
+
+      - name: Check formatting
+        if: matrix.java == 'temurin@8'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' scalafmtCheckAll 'project /' scalafmtSbtCheck
+
+      - name: scalaJSLink
+        if: matrix.project == 'rootJS'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' Test/scalaJSLinkerResult
+
+      - name: nativeLink
+        if: matrix.project == 'rootNative'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' Test/nativeLink
+
+      - name: Test
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' test
+
+      - name: Check binary compatibility
+        if: matrix.java == 'temurin@8'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' mimaReportBinaryIssues
+
+      - name: Generate API documentation
+        if: matrix.java == 'temurin@8'
+        run: sbt 'project ${{ matrix.project }}' '++${{ matrix.scala }}' doc
+
+  coverage:
+    name: Generate coverage report
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        scala: [2.12.15, 2.13.7]
+        java: [temurin@11]
```

Comment:
If coverage is disabled on 2.12 then it doesn't make sense to add a CI job for it.

https://github.com/circe/circe-generic-extras/blob/043123d97498f6b8fdbe0294b8ba968d9ee9f8f4/build.sbt#L85-L86

Lots of little details here. I look forward to getting it all right once in a shared plugin :)

Reply:
Fixed.